test/cppsuite/tests/session_microbenchmarks.cpp

```diff
@@ -0,0 +1,230 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "src/util/instruction_counter.h"
+#include "src/common/constants.h"
+#include "src/common/logger.h"
+#include "src/main/test.h"
+
+namespace test_harness {
+/* Defines what data is written to the tracking table for use in custom validation. */
+class operation_tracker_session_microbenchmarks : public operation_tracker {
+
+public:
+    operation_tracker_session_microbenchmarks(
+      configuration *config, const bool use_compression, timestamp_manager &tsm)
+        : operation_tracker(config, use_compression, tsm)
+    {
+    }
+
+    void
+    set_tracking_cursor(WT_SESSION *session, const tracking_operation &operation,
+      const uint64_t &collection_id, const std::string &key, const std::string &value,
+      wt_timestamp_t ts, scoped_cursor &op_track_cursor) override final
+    {
+        /* You can replace this call to define your own tracking table contents. */
+        operation_tracker::set_tracking_cursor(
+          session, operation, collection_id, key, value, ts, op_track_cursor);
+    }
+};
+
+static void
+make_insert(thread_worker *tc, const std::string &id)
+{
+    std::string cursor_uri = tc->db.get_collection(0).name;
+
+    auto cursor = tc->session.open_scoped_cursor(cursor_uri);
+    cursor->set_key(cursor.get(), "key" + id);
+    cursor->set_value(cursor.get(), "value1" + id);
+    int ret = cursor->insert(cursor.get());
+
+    if (ret != 0) {
+        logger::log_msg(
+          LOG_ERROR, "session_microbenchmarks: expected insert id " + id + " to succeed");
+    }
+}
+
+/*
+ * Class that defines operations that do nothing as an example. This shows how database operations
+ * can be overridden and customized.
+ */
+class session_microbenchmarks : public test {
+public:
+    session_microbenchmarks(const test_args &args) : test(args)
+    {
+        init_operation_tracker(
+          new operation_tracker_session_microbenchmarks(_config->get_subconfig(OPERATION_TRACKER),
+            _config->get_bool(COMPRESSION_ENABLED), *_timestamp_manager));
+    }
+
+    void
+    run() override final
+    {
+        /* You can remove the call to the base class to fully customize your test. */
+        test::run();
+    }
+
+    void
+    background_compact_operation(thread_worker *) override final
+    {
+        logger::log_msg(LOG_WARN, "background_compact_operation: nothing done");
+    }
+
+    void
+    checkpoint_operation(thread_worker *) override final
+    {
+        logger::log_msg(LOG_WARN, "checkpoint_operation: nothing done");
```

Comment:
Update to be something like. "checkpoint_operation: not done as this is a performance test".